lib/swm050/clk.c

```diff
@@ -0,0 +1,117 @@
+/** @addtogroup clk_file Clock peripheral API
+ * @ingroup peripheral_apis
+ * LGPL License Terms @ref lgpl_license
+ * @author @htmlonly &copy; @endhtmlonly 2019
+ * Caleb Szalacinski <contact@skiboy.net>
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2019 Caleb Szalacinski <contact@skiboy.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**@{*/
+#include <libopencm3/swm050/clk.h>
+#include <libopencm3/swm050/sysctl.h>
+
+
+/*---------------------------------------------------------------------------*/
+/** @brief Change the system clock multiplier and divider
+
+Change system clock speed.
+
+@param[in] mhz Base clock speed @ref clk_speed
+	     The base clock speed, before the clock divider
```

Comment:
I've got to say, a bool here feels like a very non-obvious argument type to a user.  espectially when the description is "the speed"  Could you explain this a bit more?  Maybe just to me, maybe it's obvious to the end user, but it feels like a clunky api like this.

Reply:
There are 2 possible states for the clock multiplier, stored in a bit.  the clk_speed defines provide a way for the user to choose one of these.  It is an odd type, but there are only 2 possible options.

Reply:
still, the values aren't "truthy" and "falsey" they're an enum of length 2.  I at least find the idea of a bool for this argument confusing and non-obvious.

Reply:
Fixed.